### DISCUSSION TITLE
I want to fire multiple tasks inside a task: Is `group(task1.s(),task2.s()...)` the best tool for this?

### DISCUSSION THREAD

**Discussion by abceleung** [celery/celery#9063] [Q&A]
I am reading the [Avoid launching synchronous subtasks](https://docs.celeryq.dev/en/latest/userguide/tasks.html#avoid-launching-synchronous-subtasks) section in the docs, but I am not sure I understand it. 

My flow is like this:
1. Flask application fires master task
2. Master task fires multiple independent subtasks (can be more than the count of celery workers. Is this bad?) and wait for their result
3. Master task fires another set of independent subtasks, dependent on the result of step 2

Is this a good approach? Am I "launching synchronous subtasks"?

```python
from celery import group

@app.task()
def subtask(index):
    time.sleep(3)
    if index % 2 == 0:
        return 'hi'
    return 'bye'

@app.task()
def master_task(subtask_count):
    subtasks1 = group(subtask.s(i) for i in range(0, subtask_count))()
    result1 = subtasks1.get()
    subtasks2 = group(subtask.s(i) for i in result1 if i == 'hi')()
    result2 = subtasks2.get()
    return result2
```

**Comment by thedrow** [comment]
You shouldn't call `subtasks1.get()` inside a task. This may block Celery indefinitely.
Instead you should do the following:
```python
@app.task(bind=True)
def master_task2(self, result1):
  self.replace(group(subtask.s(i) for i in result1 if i == 'hi'))

@app.task(bind=True)
def master_task(self, subtask_count):
   self.replace(group(subtask.s(i) for i in range(0, subtask_count)) | master_task2.s())
```

See [documentation ](https://docs.celeryq.dev/en/stable/reference/celery.app.task.html#celery.app.task.Task.replace) for details.
